web/src/features/PreflightChecks/api/getPreflightResult.tsx

```diff
@@ -0,0 +1,137 @@
+import { useQuery } from "react-query";
+import { Utilities } from "@src/utilities/utilities";
+import { PreflightCheck, PreflightResponse } from "../types";
+import { useState } from "react";
+
+async function getPreflightResult({
+  apiEndpoint = process.env.API_ENDPOINT,
+  slug,
+  sequence,
+}: {
+  apiEndpoint?: string;
+  slug: string;
+  sequence?: number;
+}): Promise<PreflightResponse> {
+  const getUrl = sequence
+    ? `${apiEndpoint}/app/${slug}/sequence/${sequence}/preflight/result`
+    : `${apiEndpoint}/app/${slug}/preflight/result`;
+  const jsonResponse = await fetch(getUrl, {
+    method: "GET",
+    headers: {
+      Authorization: Utilities.getToken(),
+    },
+  });
+
+  if (!jsonResponse.ok) {
+    throw new Error(
+      `Encountered an error while fetching preflight results: Unexpected status code: ${jsonResponse.status}`
+    );
+  }
+
+  const response: PreflightResponse = await jsonResponse.json();
+
+  // unmarshall these nested JSON strings
+  if (typeof response?.preflightResult?.result === "string")
+    response.preflightResult.result = JSON.parse(
+      response.preflightResult.result
+    );
```

Comment:
wonder if using axios would be helpful here since axios responses are already served as javascript object. You wouldn't need to parse.

Reply:
might try that out.